### ISSUE TITLE
MySQL 5.7.8 requires database user password for 'rails'@'localhost' user

### ISSUE THREAD

**Issue by yahonda** [rails/rails#21107]
Installed MySQL 5.7.8 then tried to create rails@localhost user for ActiveRecord unit testing then it gets this error.

```
mysql> CREATE USER 'rails'@'localhost';
ERROR 1819 (HY000): Your password does not satisfy the current policy requirements
mysql>
```

* Steps to reproduce 

```
$ rpm -qa |grep mysql-community-server
mysql-community-server-5.7.8-0.3.rc.el7.x86_64
```

```ruby
$ mysql -u root -p
Enter password:
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 2
Server version: 5.7.8-rc MySQL Community Server (GPL)

Copyright (c) 2000, 2015, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql>  SHOW VARIABLES LIKE 'validate_password%';
+--------------------------------------+--------+
| Variable_name                        | Value  |
+--------------------------------------+--------+
| validate_password_dictionary_file    |        |
| validate_password_length             | 8      |
| validate_password_mixed_case_count   | 1      |
| validate_password_number_count       | 1      |
| validate_password_policy             | MEDIUM |
| validate_password_special_char_count | 1      |
+--------------------------------------+--------+
6 rows in set, 1 warning (0.00 sec)

mysql> CREATE USER 'rails'@'localhost';
ERROR 1819 (HY000): Your password does not satisfy the current policy requirements
mysql>
```

* Workaround is setting /etc/my.conf using `validate-password = off`

```
$ mysql -u root -p
Enter password:
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 2
Server version: 5.7.8-rc MySQL Community Server (GPL)

Copyright (c) 2000, 2015, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql>  SHOW VARIABLES LIKE 'validate_password%';
Empty set, 1 warning (0.00 sec)

mysql> CREATE USER 'rails'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> GRANT ALL PRIVILEGES ON activerecord_unittest.*
    -> to 'rails'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> GRANT ALL PRIVILEGES ON activerecord_unittest2.*
    -> to 'rails'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> GRANT ALL PRIVILEGES ON inexistent_activerecord_unittest.*
    -> to 'rails'@'localhost';
Query OK, 0 rows affected (0.00 sec)

mysql> exit
```

It would be introduced by this change. 

http://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-8.html

```
The validate_password plugin was not installed by RPM packages for platorms using systemd or SysV-style initialization scripts. (Bug #18438833)
```
I do not know what is the better way for ActiveRecord testing yet, disable this feature or setting some default password for 'rails'@'localhost' user.

**Comment by bronzle** [comment]
You can modify `activerecord/test/config.yml` -- this file is not followed by git, you can add a password there if you need to, if the file doesn't exist, copy `activerecord/test/config.example.yml` and rename it to 
`activerecord/test/config.yml`

See: https://github.com/rails/rails/blob/master/activerecord/RUNNING_UNIT_TESTS.rdoc

**Comment by yahonda** [comment]
Thanks for the info. I overlooked that we can set password for ActiveRecord unit tests. let me try.

**Comment by yahonda** [comment]
I have tested using `activerecord/test/config.yml`. Then found some problems can be fixed by #21288. 
Closing this since my original question has been answered. Thanks @bronzle

**Comment by wangyouyan** [comment]
Actually,It's so easy to solve this issue.
firstly,we can use 'ps -ef |grep mysql' and get the below result:
ps -ef |grep mysql
-> 74 1389 1 0 5:03下午 ?? 0:00.27 /usr/local/mysql/bin/mysqld --user=_mysql --basedir=/usr/local/mysql --datadir=/usr/local/mysql/data --plugin-dir=/usr/local/mysql/lib/plugin --log-error=/usr/local/mysql/data/mysqld.local.err --pid-file=/usr/local/mysql/data/mysqld.local.pid --port=3307

Second,we entry the 'usr/local/mysql/lib/plugin',get some .so file as below:
adt_null.so libpluginmecab.so mypluglib.so semisync_slave.so
debug libtest_framework.so mysql_no_login.so validate_password.so
ha_example.so libtest_services.so rewrite_example.so version_token.so
innodb_engine.so libtest_services_threaded.so rewriter.so
libmemcached.so locking_service.so semisync_master.so

finally,delete this file 'validate_password.so '
sudo rm -rf validate_password.so

Don't fogot to restart the mysql service.
